unstructured/partition/email.py

```diff
@@ -83,6 +83,21 @@ def _parse_received_data(data: str) -> list[Element]:
 
 
 def _parse_email_address(data: str) -> tuple[str, str]:
+    """Extract and return the name and email address from the given email header.
+
+    This function assumes that the email address is enclosed in angle brackets and follows
+    the format specified by the PATTERN regex.
```

Comment:
I'm not sure this is quite a safe assumption, is it? An email address can also appear as:
```
From: steve@unstructured.io
```
can't it?

Also, I'm pretty sure that strings with spaces in them need to be quoted by themselves, like the "John Doe" nickname in `"John Doe" <john.doe@example.com>`.

I think we're going to want a separate parameterized unit test for `_parse_email_address()` that has test cases for all the varieties we can think of. The variations would be with and without quotes, with and without nickname+<mailbox>, and maybe a few others.

Reply:
Btw, parsing this is probably a little too much to ask of regular expressions. I'd look into this stdlib utility method as an alternative: https://docs.python.org/3/library/email.utils.html#email.utils.parseaddr

I'm sure they've given it more thought and testing than we'd like to invest ourselves :)

Reply:
Whether or not we can assume angle brackets and the likes depends on the standard for email headers I think. This function is only called by `partition_email_header` when iterating through `Message.raw_items()`, which has seemed like it will always have the angle brackets, but I'll look into it

Reply:
@MthwRobinson and @scanny, I figured out how to use `email` to get addresses more reliably. In doing so, I noticed that we are returning the full header value (display_name and email address) for `From` and `To` and am wondering if we want to make these just the email address or keep them as they are. We've been talking about just returning the email address for the `Cc` and `Bcc` fields, so I'm wondering if we want to treat all the addresses the same way or not.

```
E       AssertionError: assert {'email_messa.../rfc822', ...} == {'email_messa.../rfc822', ...}
E
E         Omitting 7 identical items, use -vv to show
E         Differing items:
E         {'sent_from': 'mrobinson@unstructured.io'} != {'sent_from': ['Matthew Robinson <mrobinson@unstructured.io>']}
```

Reply:
`Cc` and `Bcc` should be consistent with the current behavior for `From` and `To`

Reply:
I'd be inclined to include the nickname when present. I suppose the rationale for me is:
- removing it would be "destructive"
- removing it would make the field less human-readable

I'd say whatever we take out though ought to be round-trippable. Like still a valid RFC822 email address such that you can give that string to `email.parse` or whatever and it can reliably parse it as an email address.